source/extensions/common/aws/signer_base_impl.h

```diff
@@ -24,6 +24,22 @@ class SignatureHeaderValues {
 
 using SignatureHeaders = ConstSingleton<SignatureHeaderValues>;
 
+class SignatureQueryParameterValues {
+public:
+  // Query string parameters require camel case
+  const std::string AmzAlgorithm{"X-Amz-Algorithm"};
+  const std::string AmzCredential{"X-Amz-Credential"};
+  const std::string AmzDate{"X-Amz-Date"};
+  const std::string AmzRegionSet{"X-Amz-Region-Set"};
+  const std::string AmzSecurityToken{"X-Amz-Security-Token"};
+  const std::string AmzSignature{"X-Amz-Signature"};
+  const std::string AmzSignedHeaders{"X-Amz-SignedHeaders"};
+  const std::string AmzExpires{"X-Amz-Expires"};
+  const uint16_t DefaultExpiration = 5;
```

Comment:
nit: do you want to add code comments to indicate the unit as seconds?

Reply:
addressed